### ISSUE TITLE
Allow reordering for incomplete cholesky

### ISSUE THREAD

**Issue by learning-chip** [JuliaLinearAlgebra/Preconditioners.jl#30]
Reordering can reduce fill-in for both complete and incomplete factorizations. The `CholeskyPreconditioner` calls `lldl()` in `LimitedLDLFactorizations.jl`, but without suppling the reorder parameter:

https://github.com/JuliaLinearAlgebra/Preconditioners.jl/blob/52b3702e9c8c5d42d6092ed2df7efdbe07c81411/src/incompletecholesky.jl#L8-L13

[In lldl's example](https://github.com/JuliaSmoothOptimizers/LimitedLDLFactorizations.jl/blob/main/examples/example.jl), AMD or METIS reordering can be used like:
```julia
AMD_P = amd(K1)
METIS_P, METIS_invP = Metis.permutation(K)
...
LLDL, t, b, g, m = @timed lldl(K1, Kdiag, AMD_P, memory = p)
...
LLDL, t, b, g, m = @timed lldl(K1, Kdiag, METIS_P, memory = p)
```

**Comment by mohdibntarek** [comment]
A PR would be welcome :)

**Comment by mohdibntarek** [comment]
Fixed by #32.
